app/js/user.js

```diff
@@ -88,6 +87,49 @@
           });
         });
       });
+    },
+
+    /**
+     * Updates the local user data as well as last sync time.
+     * @param {object} user_data - The ammended user data and settings i.e.
+     * user_data = {
+     *   user: user,
+     *   settings, settings
+     * };
+     * @returns The updated user object from indexedDB
+     */
+    update_user: function(user_data) {
+      SumoDB.update_user(user_data).then(function(user) {
+        user.last_sync = Date.now();
+        return asyncStorage.setItem('user', user).then(function() {
+          return user;
+        });
+      });
+    },
+
+    /**
+    * Sets or updates the specific user preference on the server and locally.
+    * @param {string} pref - The preference to set
+    * @param {boolean} status - The status of the preference.
+    */
+    set_preference: function(pref, status) {
```

Comment:
I was waiting on feedback on the above so that I could address all three in one commit.